Terminal.Gui/Core/MainLoop.cs

```diff
@@ -73,7 +78,7 @@ public sealed class Timeout {
 		/// <summary>
 		/// Gets the list of all idle handlers.
 		/// </summary>
-		public List<Func<bool>> IdleHandlers => idleHandlers;
+		public List<Func<bool>> IdleHandlers => new List<Func<bool>>(idleHandlers);
```

Comment:
`IdleHandlers` is a public property that was introduced in 68b40a93f11335651ee8877f812ee27b0b9f64f1.

This PR changes the property to return a new list every time which means it is now effectively immutable (which is probably a good thing).  But it doesn't look immutable to an API user.  Should we change this to:

```csharp
/// <summary>
/// Gets the list of all idle handlers.
/// </summary>
public ReadOnlyCollection<Func<bool>> IdleHandlers => idleHandlers.AsReadOnly();
```

Its a breaking change but quite minor one and to a new bit of code recently introduced.  Also I think this is only going to be used by WebConsole so far.

Reply:
I agree. It must be read only and return the idlehandlers field and not a new list.

Reply:
How about write a unit test trying changing the public `IdleHandlers` field to be sure that the private `idleHandlers` field doesn't changes at all outside the Terminal.Gui namespace?.

Reply:
I've made that changes as suggested by tznind - that makes sense to me.

BDisp: I'm not sure how we would create a unit test for this as it's now a compile time error to attempt to change IdleHandlers. Could you elaborate further?

Reply:
> BDisp: I'm not sure how we would create a unit test for this as it's now a compile time error to attempt to change IdleHandlers. Could you elaborate further?

Sorry @montekarlos, you are right it already throws at compile time. Forget that and thanks for your important contribution.

Reply:
@BDisp @tznind Hey is there anything else I need to do with this PR?

Reply:
I don't think so. We just have to wait for @tig to have time available to check out.

Reply:
Just reading up on the docs on `AsReadOnly()` and noticed this:

> However, if changes are made to the underlying [List<T>](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=net-6.0) object, the read-only collection reflects those changes.
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.asreadonly?view=net-6.0

So I think we should actually have `new List<Func<bool>>(idleHandlers).AsReadOnly();`.  That way it will be properly immutable.  Also I wondered if we should have `lock (idleHandlersLock) ` around it too?

So it would be:

```csharp
/// <summary>
/// Gets the list of all idle handlers.
/// </summary>
public ReadOnlyCollection<Func<bool>> IdleHandlers {
	get {
		lock (idleHandlersLock) {
			return new List<Func<bool>> (idleHandlers).AsReadOnly();
		}
	}
}
```

Reply:
Agreed